models/agent_helpers.go

```diff
@@ -45,8 +45,14 @@ func checkUniqueAgentID(q *reform.Querier, id string) error {
 }
 
 // FindAllAgents returns all Agents.
-func FindAllAgents(q *reform.Querier) ([]*Agent, error) {
-	structs, err := q.SelectAllFrom(AgentTable, "ORDER BY agent_id")
+func FindAllAgents(q *reform.Querier, agentType *AgentType) ([]*Agent, error) {
+	var conditions []string
+	var args []interface{}
+	if agentType != nil {
+		conditions = append(conditions, "WHERE agent_type = $1")
+		args = append(args, agentType)
+	}
+	structs, err := q.SelectAllFrom(AgentTable, fmt.Sprintf("%s ORDER BY agent_id", strings.Join(conditions, " AND ")), args...)
```

Comment:
`conditions = append(conditions, "WHERE agent_type = $1")` in couple with `strings.Join(conditions, " AND ")` looks strange - cannot be multiple `WHERE` in query.

Reply:
No, it's not possible, at least for now. Made it this way to have the same structure with queries in other helpers.